target/linux/ramips/dts/mt7621_cudy_wr1300.dts

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "cudy,wr1300", "mediatek,mt7621-soc";
+	model = "CUDY WR1300";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	aliases {
+		led-boot = &led_sys;
+		led-failsafe = &led_sys;
+		led-running = &led_sys;
+		led-upgrade = &led_sys;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_sys: status {
+			label = "wr1300:green:status";
+			gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
+		};
+
+		wps {
+			label = "wr1300:green:wps";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		usb {
+			label = "wr1300:green:usb";
+			gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;
+			trigger-sources = <&xhci_ehci_port1>, <&ehci_port2>;
+			linux,default-trigger = "usbport";
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Can this go faster?

Reply:
This is what was in the dts provided by the manufacturer for OpenWRT (See the for developers zip here: http://www.cudytech.com/filedownload/284029). Is it worth testing what happens if it is clocked faster? If so, could you direct me to documentation on how to do this, or is it simply trial and error to see if images boot or crash?

Reply:
Well, 10 MHz is just a value everybody copies without thinking. Actually, the devices can typically go much faster which boosts read speed substantially. Typically, you can look into same spec sheets if available and/or try what's reasonably possible. Apart from that, you may also check other/similar supported devices, of course.

Reply:
Tested at 104 MHz as per the datasheet https://www.winbond.com/resource-files/w25q128fv%20rev.l%2008242015.pdf

The router boots. Is there any specific stability issue to search for/test?

Reply:
The datasheet says 104MHz is not for READ, so you should have used `m25p,fast-read`... But it seems the device does not complain without it, I had also noticed that phenomenon on my devices in the past. I don't know why, I haven't check it with a logic analyzer. Perhaps the datasheet is wrong, or perhaps the hardware actually uses a lower frequency.

Anyway, if it works, then no problem with it, good. But my concern is that the situation might be different on someone else's device. (See #3159) Unless you are the manufacturer, you can't really be sure if the freq works on all devices. And if the flash fails, it's really hard for a common user to fix it as it'll need a serial access. So I believe we'd better take a conservative approach here.

MT7621 supports only a set of freqs, `sys_freq/n` where sys_freq = 220 and n >= 2. If you set `spi-max-frequency` to 104MHz, the actual freq used in hardware is 73.3MHz. So I suggest setting `spi-max-frequency` to 80MHz (with `m25p,fast-read`), or 50MHz to be safe. You can test the flash speed by `time dd if=/dev/mtdblock3 of=/dev/null bs=64k`.

Reply:
Changed to 80 MHZ with the fast read. Output of benchmark is 
`root@OpenWrt:~# time dd if=/dev/mtdblock3 of=/dev/null bs=64k
248+0 records in
248+0 records out
real	0m 3.56s
user	0m 0.00s
sys	0m 3.51s
root@OpenWrt:~# time dd if=/dev/mtdblock3 of=/dev/null bs=64k
248+0 records in
248+0 records out
real	0m 3.57s
user	0m 0.00s
sys	0m 3.53s
root@OpenWrt:~# 
`